### PR TITLE
Declare as TempSensor, Light, and ColorControl

### PR DESCRIPTION
**Pull request by gcohler**
This is pretty trivial so far.  Declares device as a temp sensor and a color control.  

I will also try to add an on/off switch that controls the light.  But that will take me a bit longer.

### CODE REVIEW

File: pkg/sense_hat_adapter.py [hunk 1]
```diff
@@ -144,7 +144,6 @@ def update(self):
         elif self.name == 'temperature':
             value = self.device.controller.temperature
         else:
-            print("warning: %s not handled" % self.name)
```
Comment (rzr) [inline pkg/sense_hat_adapter.py]: why remove this line ?

File: pkg/sense_hat_adapter.py [hunk 2]
```diff
@@ -64,7 +64,9 @@ def __init__(self, adapter):
         self.name = 'Sense Hat'
         self.name = 'SenseHat'
         self.description = 'Expose SenseHat sensors and actuators'
-        self._type = ['TemperatureSensor', 'ColorControl']
+        self._type = ['TemperatureSensor', 'ColorControl', 'Light']
+        self.color = '#ffffff'
+        self.onOff = True
```
Comment (rzr) [inline pkg/sense_hat_adapter.py]: I would initialize to False but may not be critical

File: pkg/sense_hat_adapter.py [hunk 3]
```diff
@@ -106,7 +108,17 @@ def __init__(self, adapter):
                     'type': 'string',
                     'readOnly': False
                 },
-                '#ffffff')
+                self.color)
+            self.properties['onOff'] = SenseHatProperty(
```
Comment (rzr) [inline pkg/sense_hat_adapter.py]: 'on'
is commonly used

File: pkg/sense_hat_adapter.py [hunk 4]
```diff
@@ -167,14 +179,22 @@ def update(self):
             self.set_cached_value(value)
             self.device.notify_property_changed(self)
 
-    def set_value(self, value):
-        if self.name == 'message':
-            self.device.controller.show_message(value)
-        elif self.name == 'color':
+    def set_value(prop, value):
```
Comment (rzr) [inline pkg/sense_hat_adapter.py]: self is the object that method applies too, keep it or justify why it should be renamed

File: pkg/sense_hat_adapter.py [hunk 5]
```diff
@@ -190,7 +190,7 @@ def set_value(prop, value):
         elif prop.name == 'onOff':
             if value == False:
                 prop.device.controller.clear([0, 0, 0])
-            else
+            else:
```
Comment (rzr) [inline pkg/sense_hat_adapter.py:269]: change to be squashed

File: pkg/sense_hat_adapter.py [hunk 6]
```diff
@@ -108,7 +106,7 @@ def __init__(self, adapter):
                     'type': 'string',
                     'readOnly': False
                 },
-                self.color)
+                '#ffffff')
```
Comment (rzr) [inline pkg/sense_hat_adapter.py:128]: may reference to adapter used instead , use use self.device.color

File: pkg/sense_hat_adapter.py [hunk 7]
```diff
@@ -183,17 +181,22 @@ def set_value(prop, value):
...
+            self.set_cached_value(value)
         elif prop.name == 'onOff':
             if value == False:
+                print("info: Turning light off.")
                 prop.device.controller.clear([0, 0, 0])
             else:
-                color = [int(self.color[1:3], 0x10),
-                         int(self.color[3:5], 0x10),
-                         int(self.color[5:7], 0x10)]
+                colorString = self.properties.color.value
```
Comment (rzr) [inline pkg/sense_hat_adapter.py]: isnt self a property here ?

File: pkg/sense_hat_adapter.py [hunk 8]
```diff
@@ -177,27 +177,27 @@ def update(self):
             self.set_cached_value(value)
             self.device.notify_property_changed(self)
 
-    def set_value(prop, value):
-        if prop.name == 'message':
-            prop.device.controller.show_message(value)
-        elif prop.name == 'color':
+    def set_value(self, value):
```
Comment (rzr) [inline pkg/sense_hat_adapter.py:256]: ok fine 
try to rebase and squash changes

**Review by rzr — CHANGES_REQUESTED** [review]

**Comment by rzr** [comment]
I am thinking wouldn't off (clear([0,0,0]) conflict with color property ? well if color is stored it can be restored if on is triggered again.

But setting 'message' should be "unshown" unless 'on' property is true but message text updated , for later 'on'.

File: pkg/sense_hat_adapter.py (hunk 1 above)
Comment (gcohler) [reply pkg/sense_hat_adapter.py]: Just was adding lots of stuff in the log.  But up to you.  I don't really understand, but the logging sometimes works and sometimes doesn't.

**Comment by gcohler** [comment]
Ok.  @rzr:  I think this is ready to merge.  Thank you for your patience.

File: pkg/sense_hat_adapter.py [hunk 9]
```diff
@@ -121,12 +121,21 @@ def __init__(self, adapter):
...
                 {
                     '@type': 'ColorProperty',
-                    'label': "Color",
+                    'label': "Light Color",
                     'type': 'string',
                     'readOnly': False
                 },
                 '#ffffff')
-
+            self.properties['lightOn'] = SenseHatProperty(
```
Comment (rzr) [inline pkg/sense_hat_adapter.py]: I think "on" is better here as there is only one "switch"

File: pkg/sense_hat_adapter.py [hunk 10]
```diff
@@ -121,12 +121,21 @@ def __init__(self, adapter):
...
+                self,
+                "lightOn",
```
Comment (rzr) [inline pkg/sense_hat_adapter.py]: "on"

File: pkg/sense_hat_adapter.py [hunk 11]
```diff
@@ -245,13 +254,28 @@ def update(self):
...
         if self.name == 'message':
             self.device.controller.show_message(value)
         elif self.name == 'color':
-            color = [int(value[1:3], 0x10),
-                     int(value[3:5], 0x10),
-                     int(value[5:7], 0x10)]
-            self.device.controller.clear(color)
+            if self.device.properties['lightOn'].value:
+                color = [int(value[1:3], 0x10),
+                         int(value[3:5], 0x10),
```
Comment (rzr) [inline pkg/sense_hat_adapter.py:263]: I will update color even if off

File: pkg/sense_hat_adapter.py [hunk 12]
```diff
@@ -245,13 +254,28 @@ def update(self):
...
+                         int(value[5:7], 0x10)]
+                self.device.controller.clear(color)
```
Comment (rzr) [inline pkg/sense_hat_adapter.py:265]: I would use this only for 'on' changes

File: pkg/sense_hat_adapter.py [hunk 13]
```diff
@@ -245,13 +254,28 @@ def update(self):
...
+            else:
+                colorString = self.device.properties['color'].value
+                color = [int(colorString[1:3], 0x10),
+                         int(colorString[3:5], 0x10),
+                         int(colorString[5:7], 0x10)]
+                self.device.controller.clear(color)
         else:
             print("warning: %s not handled" % self.name)
             return
+        if value != self.value:
```
Comment (rzr) [inline pkg/sense_hat_adapter.py:278]: ok

**Review by rzr — CHANGES_REQUESTED** [review]
looks like a great improvement please keep "on" for now

File: pkg/sense_hat_adapter.py (hunk 10 above)
Comment (gcohler) [reply pkg/sense_hat_adapter.py]: Done

File: pkg/sense_hat_adapter.py (hunk 9 above)
Comment (gcohler) [reply pkg/sense_hat_adapter.py]: done

File: pkg/sense_hat_adapter.py (hunk 11 above)
Comment (gcohler) [reply pkg/sense_hat_adapter.py:263]: It updates the color at line 279.  But it doesn't turn on the light if it is off.

File: pkg/sense_hat_adapter.py (hunk 12 above)
Comment (gcohler) [reply pkg/sense_hat_adapter.py:265]: Agreed.

File: pkg/sense_hat_adapter.py (hunk 11 above)
Comment (gcohler) [reply pkg/sense_hat_adapter.py:263]: There's no real on-off.  Setting the color to 0,0,0 is off.  Setting it to non-0.0.0 is on.

**Comment by gcohler** [comment]
Ok.  Look again now.

**Review by rzr — APPROVED** [review]
Ok I can merge but I can hold for some improvements try this sequeence : set off , set color ; set on  , will it work ?

I can refactor this later if it helps

**Comment by gcohler** [comment]
> Ok I can merge but I can hold for some improvements try this sequeence : set off , set color ; set on , will it work ?

Yes.  it works just fine.  I save the color string (e.g. #123456) in the property.  Then decompose it only when changing the light.

**Comment by gcohler** [comment]
So whenever the color changes, it is saved.  And whenever the light goes from on to off, it turns the light to 0, 0, 0.  Then whenever the light goes from off to on, it goes to its old color.  And if the light is on and changes color, then it also changes to the new color.

**Pull request merged**